nodejs/aws-infra/experimental/ecs/cluster.ts

```diff
@@ -160,10 +179,16 @@ export interface ClusterArgs {
     name?: pulumi.Input<string>;
 
     /**
-     * The security group to place new instances into.  If not provided, a default will be
-     * created.
+     * The security group to place new instances into.  If neither [securityGroups] nor
+     *  [securityGroupIds] are provided, a default will be created for the Cluster.
      */
     securityGroups?: x.ec2.SecurityGroup[];
+
+    /**
+     * The security group ids to place new instances into.  If neither [securityGroups] nor
+     *  [securityGroupIds] are provided, a default will be created for the Cluster.
+     */
+    securityGroupIds?: pulumi.Input<string>[];
```

Comment:
Any reason not to just have one field and allow it to be `securityGroups?: Array<x.ec2.SecurityGroup | pulumi.Input<string>>;`?  That would make the API a little simpler, and would even seem to make the implementation here a bit simpler.  Lastly it's closer to how our strongly typed properties look on raw AWS APIs when we add `AltType` annotations (although I don't think we have cases of arrays of these alternatives).

Reply:
@lukehoban I'll try that.  i personally have a bias against those sorts of complex signatures as i find them harder to read.  Also, impl-wise it is a bit of a PITA as you have to dynamically check for 'string/promise/output'.  But if you honestly prefer it, i'm ok with that :)  let me create an alternate pr and we can compare!